git/setup.rst

```diff
@@ -111,18 +111,44 @@ To have your credentials cached for 1 hour (3600 seconds):
 
    git config --global credential.helper store
 
 Once a credential helper is enabled, the next time you ``git push``, you will add your credentials to the helper.
 
-Remember that if you have 2FA enabled, you will create and use a `personal access token <https://github.com/settings/tokens>`_ instead of your GitHub password.
+If you are using HTTPS access, you will need to create and use a :ref:`personal access token <git-auth-tokens>` instead of your GitHub password, as described below.
 
-If you find that ``git push`` is not working but also not asking you for credentials, you may need to manually insert the username/password or token into the credential store or macOS keychain.
+.. _git-auth-tokens:
 
-The DM Git LFS documentation has further information about :ref:`authenticating with our LFS storage backend <git-lfs-auth>`.
+Git authentication tokens
+-------------------------
+
+Due to GitHub's authentication interface, you must use a personal access token instead of a password when pushing to GitHub via HTTPS.
+GitHub access via ssh does not require a token, but :doc:`Git LFS repos <git-lfs>` work best via HTTPS.
+A credential helper, as :ref:`described above <git-credential-helper>`, will let you cache this token, so you do not have to save it (you can only view it on GitHub when you generate it).
+Create a personal token at `GitHub Personal Access Tokens <https://github.com/settings/tokens>`_, using the "generate new token" button in the upper right, with the "classic" interface.
+Give it a name (e.g. "LSST USDF"), choose an expiration date (when the token expires you will have to generate a new one and re-enter it into your credential cache), and set the following permissions::
+
+  repo->public_repo
+  admin:org->read:org
+
+!!! NOTE: do we also need ``workflow``, to be able to change GitHubActions, or any others? Should we have a note about other options that people may want?
+
+The next time you run a Git command that requires authentication, Git will ask you to authenticate with both GitHub (for the push via HTTPS) and, if you are pushing to a Git LFS repo, with LSST's Git LFS server (for authentication of the LFS upload)::
+
+   Username for 'https://github.com': <GitHub username>
+   Password for 'https://<user>@github.com': <GitHub token>
+   Username for 'https://git-lfs.lsst.codes': <GitHub username>
+   Password for 'https://<user>@git-lfs.lsst.codes': <GitHub token>
+
+At the prompts, enter your GitHub username and token.
+
+Once your credentials are cached, you won't need to repeat this process on your system (:ref:`unless you opted for the cache-based credential helper <git-credential-helper>`) until the token expires.
+
+If you find that ``git push`` is not working but also not asking you for credentials, you may need to manually insert the username and token into the credential store or macOS keychain. NOTE: we should expand on this, as I recall having this problem but don't know what I did to fix it.
```

Comment:
On macOS I think I had to run up Keychain Access and find the token and delete it so it would prompt me again.